C# Azure AI Projects SDK for new 1DP endpoint

sophia-ramsey opened the pull request:
Emitted C# code from Typespec using feature/azure-ai-projects-1dp

Commit:
adding agents v2

Commit:
updating to latest typespec

Commit:
adding persistent agent deletion to sample

Commit:
adding conveniency functions and samples that use them

Commit:
update to sample

sophia-ramsey committed:
Custom code + samples for datasets and deployments

jsquire commented on samples/AgentSampleApp/AgentSampleApp.csproj:
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
```
This is not the correct place for this.   This `samples` area is for end-to-end samples that use multiple libraries.   Please move this into the `/samples` space in your library.

nick863 committed:
Add ci/cd config (#49469)

azure-sdk commented:
**API change check**

API changes are not detected in this pull request.

sophia-ramsey committed:
Regenerate code + add custom code (#49523)

* Custom code + samples for datasets, deployments, connections
* Generated based on [commit 37e1a52a30c527b6ad2d7ef09a9c82ad4491e8a2 ](https://github.com/Azure/azure-rest-api-specs/commit/37e1a52a30c527b6ad2d7ef09a9c82ad4491e8a2 )

sophia-ramsey committed:
remove sample app (#49625)

jsquire commented on sdk/core/Azure.Core.TestFramework/src/Azure.Core.TestFramework.csproj:
```diff
@@ -7,6 +7,8 @@
   <ItemGroup>
     <PackageReference Include="Azure.Core" />
     <PackageReference Include="Azure.Identity" />
+    <PackageReference Include="Azure.AI.Inference" />
```
Please revert.  These packages should not be dependencies of the test framework.

sophia-ramsey committed:
update sample files (#49635)

* remove old sample app

* add sample files

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.AI.Projects.OneDP](https://apiview.dev/Assemblies/Review/0a26eab939b74053850ee9f9db6e36bb?revisionId=a535767cae2349d98acc711449bdbdec)

sophia-ramsey committed:
Add evaluations sample (#49729)

* Evaluations sample

sophia-ramsey committed:
Remove 1DP (#49760)

* remove old projects folder

* remove OneDP

* rename Azure.AI.Projects.OneDP

* regenerate code

* update samples for generated code

sophia-ramsey committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into feature/azure-ai-projects-1dp

sophia-ramsey committed:
regenerate code and update samples

glharper committed:
[AI] [Projects] add ConnectionProvider to AIProjectClient (#49798)

* [AI] [Projects] add ConnectionProvider base class

* add apis for connections

* api change

* add custom AIProjectClient back in

* remove moved files

* add generated docs and samples

* update constructor to use ClientConnectionProvider

* update generated code

* Parity with existing Projects constructors

* update generated client class

* Add Agents sample

* use latest nuget for Azure.AI.Agents.Persistent

* uncomment thread running code for agents sample

* async sample

* update async sample

* add proper connection type

* update cache manager to correctly handle endpoint connection

glharper committed:
[AI] [Projects] Add openai sample using authenticated client (#50000)

sophia-ramsey committed:
update ci.projects_1dp.yml (#50002)

* update ci.projects_1dp.yml file

sophia-ramsey committed:
update connections and indexes samples (#50022)

Update connections and indexes samples for new function names and updated custom code

sophia-ramsey committed:
update samples in samples folder (#50023)

* update samples for consistency
* add sample files to samples folder for Agents and OpenAI extensions
* remove ci.projects_1dp.yml file (duplicate of ci.projects.yml now that the package is renamed)

sophia-ramsey committed:
remove duplicate AssemblyInfo (#50025)

sophia-ramsey committed:
regenerate code from typespec (#50027)

using commit c6f9c72658fa9039b159ef1aa6e6d0c2385ba0ee where 1dp branch was merged into main

sophia-ramsey committed:
update snippets (#50029)

* readme snippet update to pass pipelines

nick863 committed:
Fix cspell. (#50035)

* Fix cspell issue

* Fix samples

sophia-ramsey committed:
add telemetry samples and update connections sample (#50040)

* add telemetry getConnectionString sample
* update connections functions and fix typo

nick863 committed:
Fix generated code (#50075)

sophia-ramsey committed:
merge conflicts

sophia-ramsey committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into feature/azure-ai-projects-1dp

sophia-ramsey committed:
resolve merge conflicts and fix project setup

sophia-ramsey committed:
add inference samples and custom code (#50104)

* add inference samples and custom code

* address comments

* fix sample snippet

sophia-ramsey committed:
update changelog (#50107)

* changelog draft

* address comments

sophia-ramsey committed:
readme update (#50110)

* readme draft

* readme update

nick863 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into feature/azure-ai-projects-1dp

nick863 committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into feature/azure-ai-projects-1dp

nick863 committed:
Upgrade the Agents (#50106)

* Upgrade the Agents

* Fix

* Small fix

* Use beta version

* Use public version, update API

dargilco commented on .vscode/cspell.json:
```diff
@@ -197,7 +197,15 @@
+        "azureai",
+        "Gleu",
+        "gleu"
       ]
```
For next update/release - In .net repo is there only one cspell.json at the root? In Python we have those in every SDK folder... is it not the case for .net repo as well? If so in the next update perhpas move all of this to a local cspell.json that's just used for the Azure.AI.Projects folder.

dargilco commented on sdk/ai/Azure.AI.Projects/CHANGELOG.md:
```diff
@@ -3,12 +3,26 @@
+* Several changes to the `Connections` methods, including the response object (now it is simply called `Connection`)
+* `GetAzureOpenAIChatClient` now supports returning an authenticated `AzureOpenAI` ChatClient to be used with
+AI models deployed to the Project's AI Services. 
+* The class `ConnectionProperties` was renamed to `Connection`, and its properties have changed.
```
This seems somewhat of a dup of line 15. Was it the same in the Python change log?

nick863 replied:
Yes, we have it in both packages: https://github.com/Azure/azure-sdk-for-python/blame/e9bf60e66e4aa9ca346418c52b19efd4d4ff9478/sdk/ai/azure-ai-projects/CHANGELOG.md#L22

sophia-ramsey replied:
I added this line to line 15, so now it reads, "Several changes to the `Connections` methods, including the response object (now it is simply called `Connection`). The class `ConnectionProperties` was renamed to `Connection`, and its properties have changed."

dargilco commented on sdk/ai/Azure.AI.Projects/CHANGELOG.md:
```diff
@@ -3,12 +3,26 @@
+* The method `UploadFileRequest` on `AIProjectClient` had been removed, use `UploadFile` in `Datasets` instead.
+* Evaluator Ids are available using the class `EvaluatorIDs` and no longer require `Azure.AI.Evaluation` package to be installed.
+* Property `scope` on `AIProjectClient` is removed.
+* Property `Id` on Evaluation is replaced with `name`.
```
Did we have evaluations in C# before? I don't remember.

sophia-ramsey replied:
I see the evaluations code there, but there were no samples using it.

dargilco commented on sdk/ai/Azure.AI.Projects/README.md:
```diff
@@ -1,8 +1,16 @@
 # Azure AI Projects client library for .NET
 Use the AI Projects client library to:
```
This is what I have no for Python: "The AI Projects client library (in preview) is part of the Azure AI Foundry SDK and provides easy access to resources in your Azure AI Foundry Project. Use it to:"

nick863 replied:
Fixed.

dargilco commented on sdk/ai/Azure.AI.Projects/CHANGELOG.md:
```diff
@@ -3,12 +3,26 @@
 ### Breaking Changes
+* Azure AI Foundry Project endpoint is now required to construct the `AIProjectClient`. It has the form
+`https://<your-ai-services-account-name>.services.ai.azure.com/api/projects/<your-project-name>`. Find it in your AI Foundry Project Overview page. 
+* Agents are now implemented in a separate package `Azure.AI.Agents.Persistent`. Continue using the `agents` operations on the
```
Isn't it GetPersistanAgentsClien() in C#, and not just a ".agents" property?

dargilco commented on sdk/ai/Azure.AI.Projects/tests/Samples/Datasets/Sample_Datasets.cs:
```diff
@@ -0,0 +1,125 @@
+            datasets.Delete(datasetName, "1");
+            datasets.Delete(datasetName, "2");
+            #endregion
+        }
```
missing datasets.GetCredentials(). Can be added later.

jsquire commented on sdk/ai/Azure.AI.Projects/src/Azure.AI.Projects.csproj:
```diff
@@ -14,8 +14,11 @@
+    <PackageReference Include="Azure.Storage.Blobs" />
+    <PackageReference Include="Azure.AI.Inference" />
+    <PackageReference Include="Azure.AI.OpenAI" />
+    <PackageReference Include="System.ClientModel" />
```
```suggestion
```

`System.ClientModel` is pulled in transitively via Core.  In order to ensure no version mismatches, there shouldn't be a direct reference here.

sophia-ramsey replied:
Fixed

jsquire commented on sdk/ai/Azure.AI.Projects/src/Custom/AIProjectClient.cs:
```diff
@@ -15,69 +15,48 @@ namespace Azure.AI.Projects
+        /// <param name="endpoint">
+        /// Project endpoint. In the form "https://&lt;your-ai-services-account-name&gt;.services.ai.azure.com/api/projects/_project"
+        /// if your Foundry Hub has only one Project, or to use the default Project in your Hub. Or in the form
+        /// "https://&lt;your-ai-services-account-name&gt;.services.ai.azure.com/api/projects/&lt;your-project-name&gt;" if you want to explicitly
```
```csharp
        /// "https://&lt;your-ai-services-account-name&gt;.services.ai.azure.com/api/projects/&lt;your-project-name&gt;" if you want to explicitly
```

These encoded HTML characters will render exactly as written in both Intellisense and in Docs on MS Learn and will not be translated back to HTML.

jsquire commented on sdk/ai/Azure.AI.Projects/samples/README.md:
```diff
@@ -8,41 +8,24 @@ products:
+| [Sample8_Telemetry](https://github.com/Azure/azure-sdk-for-net/blob/feature/azure-ai-projects-1dp/sdk/ai/Azure.AI.Projects/samples/Sample8_Telemetry.md) | Sample for telemetry functions. |
 
-# Azure.AI.Projects Samples for Extensions
+# Azure.AI.Projects Samples for Agents and Inference Extensions
```
```suggestion
# Azure.AI.Projects samples for Agents and Inference extensions
```

README and sample content in Markdown format should follow the [Microsoft style guide](https://learn.microsoft.com/en-us/style-guide/capitalization#sentence-style-capitalization-in-titles-and-headings), which states titles and headings should be in Sentence-style capitalization.

sophia-ramsey replied:
Fixed

jsquire commented on sdk/ai/Azure.AI.Projects/samples/Sample10_InferenceEmbedding.md:
````diff
@@ -0,0 +1,53 @@
+
+## Synchronous Sample
+
+```C# Snippet:EmbeddingSync
````
PLease give your snippets names that are scoped to your library.  These names must be globally unique in the repository and general names like this have a risk of causing conflict.    For example, `Snippet:AI_Projects_EmbeddingSync` would be the recommended pattern.

sophia-ramsey replied:
Fixed

jsquire reviewed (changes requested):
The only blocking bits for my feedback are:

- Please remove System.ClientModel direct reference
- Please adjust snippet names to be more tightly scoped

nick863 committed:
Small fixes

nick863 committed:
Merge branch 'feature/azure-ai-projects-1dp' of https://github.com/Azure/azure-sdk-for-net into feature/azure-ai-projects-1dp

sophia-ramsey committed:
addressing comments (#50127)

* mostly readme / changelog changes
* updated snippet names to be package-specific
* changed the deployment sample variable names for consistency

sophia-ramsey committed:
update snippet (#50130)
